src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrCloneManager.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Darc.Models.VirtualMonoRepo;
+using Microsoft.DotNet.DarcLib.Helpers;
+using Microsoft.Extensions.Logging;
+
+#nullable enable
+namespace Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+
+public interface IVmrCloneManager
+{
+    /// <summary>
+    /// Prepares the local VMR clone by fetching from given remotes one-by-one until all requested commits are available.
+    /// </summary>
+    /// <param name="remoteUris">Remotes to fetch from one by one</param>
+    /// <param name="requestedRefs">List of refs that need to be available</param>
+    /// <param name="checkoutRef">Ref to check out at the end</param>
+    /// <returns>Path to the clone</returns>
+    Task<ILocalGitRepo> PrepareVmrAsync(
+        IReadOnlyCollection<string> remoteUris,
+        IReadOnlyCollection<string> requestedRefs,
+        string checkoutRef,
+        CancellationToken cancellationToken);
+
+    Task<ILocalGitRepo> PrepareVmrAsync(
+        string checkoutRef,
+        CancellationToken cancellationToken);
+}
+
+public class VmrCloneManager : CloneManager, IVmrCloneManager
+{
+    private readonly IVmrInfo _vmrInfo;
+    private readonly IVmrDependencyTracker _dependencyTracker;
+    private readonly ISourceManifest _sourceManifest;
+
+    public VmrCloneManager(
+        IVmrInfo vmrInfo,
+        IVmrDependencyTracker dependencyTracker,
+        ISourceManifest sourceManifest,
+        IGitRepoCloner gitRepoCloner,
+        ILocalGitClient localGitRepo,
+        ILocalGitRepoFactory localGitRepoFactory,
+        ITelemetryRecorder telemetryRecorder,
+        IFileSystem fileSystem,
+        ILogger<VmrCloneManager> logger)
+        : base(vmrInfo, gitRepoCloner, localGitRepo, localGitRepoFactory, telemetryRecorder, fileSystem, logger)
+    {
+        _vmrInfo = vmrInfo;
+        _dependencyTracker = dependencyTracker;
+        _sourceManifest = sourceManifest;
+    }
+
+    public async Task<ILocalGitRepo> PrepareVmrAsync(
+        IReadOnlyCollection<string> remoteUris,
+        IReadOnlyCollection<string> requestedRefs,
+        string checkoutRef,
+        CancellationToken cancellationToken)
+    {
+        var repo = await PrepareCloneInternalAsync(
+            Constants.VmrFolderName,
+            remoteUris,
+            requestedRefs,
+            checkoutRef,
+            cancellationToken);
+
+        _vmrInfo.VmrPath = repo.Path;
+        _vmrInfo.VmrUri = remoteUris.First();
```

Comment:
This looks like a trick. We are making the assumption that the vmr repo will be the first element in the `remoteUris` collection.

Reply:
I changed it so that it's only setting it when it's obvious